twins/redis/twins.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package redis
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-redis/redis"
+	"github.com/mainflux/mainflux/errors"
+	"github.com/mainflux/mainflux/twins"
+)
+
+const (
+	prefix = "twin"
+)
+
+// ErrRedisTwinSave indicates error while saving Twin in redis cache
+var ErrRedisTwinSave = errors.New("saving twin in redis cache error")
+
+// ErrRedisTwinIDs indicates error while geting Twin IDs from redis cache
+var ErrRedisTwinIDs = errors.New("get twin id from redis cache error")
+
+// ErrRedisTwinRemove indicates error while removing Twin from redis cache
+var ErrRedisTwinRemove = errors.New("remove twin from redis cache error")
+
+var _ twins.TwinCache = (*twinCache)(nil)
+
+type twinCache struct {
+	client *redis.Client
+}
+
+// NewTwinCache returns redis twin cache implementation.
+func NewTwinCache(client *redis.Client) twins.TwinCache {
+	return &twinCache{
+		client: client,
+	}
+}
+
+func (tc *twinCache) Save(_ context.Context, twin twins.Twin) error {
+	def := twin.Definitions[len(twin.Definitions)-1]
```

Comment:
lets check the len before access

Reply:
By design, twin always have at least an empty definition because of this line https://github.com/mainflux/mainflux/blob/master/twins/service.go#L147 In that sense, that would be a superfluous check. However, might make sense if we let twins creation happen outside Save mtethod. Not sure.